src/test/rgw/rgw_multi/tests.py

```diff
@@ -1316,3 +1317,124 @@ def test_bucket_creation_time():
+
+    # create a bucket on the first zone
+    bucket_name = gen_bucket_name()
+    log.info('create bucket zone=%s name=%s', z1.name, bucket_name)
+    bucket = zc1.conn.create_bucket(bucket_name)
+
+    system_access_key = z1.data['system_key']['access_key']
+
+    system_secret_key = z1.data['system_key']['secret_key']
+
+    system_host = "localhost:"+str(system_port)
+    system_host_2 = "localhost:"+str(system_port_2)
+
+    subprocess.call(['warp', 'put', '--host', system_host, '--access-key', system_access_key, '--secret-key', system_secret_key, '--insecure', '--duration=1m', '--obj.size=10KiB', '--concurrent=6', '--noprefix', '--noclear', '--bucket', bucket_name])
+
+    subprocess.Popen(['warp', 'put', '--host', system_host_2, '--access-key', system_access_key, '--secret-key', system_secret_key, '--insecure', '--duration=1m', '--obj.size=10KiB', '--concurrent=6', '--noprefix', '--noclear', '--bucket', bucket_name])
+
+    time.sleep(120)
+    zone_data_checkpoint(z2, z1)
+
+    assert zc2.get_bucket(bucket_name)
+
+
+@attr('workload_test')
+def test_basic_workload_with_lifecycle():
+    "THIS TEST IS STILL IN PROGRESS"
+    zonegroup = realm.master_zonegroup()
+    system_port = 8000
+    if len(zonegroup.zones) < 2:
+        raise SkipTest("test_basic_workload_with_lifecycle skipped. We're testing here with 2 or more zones in master zonegroup.")
+
+    zonegroup_conns = ZonegroupConns(zonegroup)
+    (zc1, zc2) = zonegroup_conns.rw_zones[0:2]
+
+    z1, z2 = (zc1.zone, zc2.zone)
+
+    # create a bucket on the first zone
+    bucket_name = gen_bucket_name()
+    log.info('create bucket zone=%s name=%s', z1.name, bucket_name)
+    bucket = zc1.conn.create_bucket(bucket_name)
+
+    system_access_key = z1.data['system_key']['access_key']
+
+    system_secret_key = z1.data['system_key']['secret_key']
+
+    system_host = "localhost:"+str(system_port)
+
+    subprocess.call(['warp', 'put', '--host', system_host, '--access-key', system_access_key, '--secret-key', system_secret_key, '--insecure', '--duration=1m', '--obj.size=10KiB', '--concurrent=6', '--noprefix', '--noclear', '--bucket', bucket_name])
+
+    zone_data_checkpoint(z2, z1)
+   
+    obj_prefix = 'oooo'
+
+    # create lifecycle policy
+    client = boto3.client('s3',
+            endpoint_url='http://localhost:'+str(system_port),
+            aws_access_key_id=system_access_key,
+            aws_secret_access_key=system_secret_key)
+    yesterday = datetime.date.today() - datetime.timedelta(days=1)
+    response = client.put_bucket_lifecycle_configuration(Bucket=bucket_name,
+            LifecycleConfiguration={'Rules': [
+                {
+                    'ID': 'rule1',
+                    'Expiration': {'Date': yesterday.isoformat()},
+                    'Filter': {'Prefix': obj_prefix},
+                    'Status': 'Enabled',
+                }
+            ]
+        }
+    )
+
+    # start lifecycle processing
+    admin(['lc', 'process'])
```

Comment:
Once the process finishes, this needs a listing of both sides to ensure that all the objects have been deleted.